### ISSUE TITLE
utils.fatal - Object has no method 'fatal'

### ISSUE THREAD

**Issue by pbowyer** [node-js-libs/node.io#36]
Thanks for your work on node.io, it's a fantastic tool!

I've just enabled jsdom in order to use jQuery, and now get the following error:

``` bash
$ node.io nuffield

/usr/local/lib/node_modules/node.io/lib/node.io/dom.js:58
            utils.fatal('jQuery is not installed. Run `npm install jquery`');
                  ^
TypeError: Object #<Object> has no method 'fatal'
    at [object Object].parseHtml (/usr/local/lib/node_modules/node.io/lib/node.io/dom.js:58:19)
    at [object Object].<anonymous> (/usr/local/lib/node_modules/node.io/lib/node.io/request.js:108:18)
    at /usr/local/lib/node_modules/node.io/lib/node.io/request.js:217:25
    at /usr/local/lib/node_modules/node.io/lib/node.io/request.js:379:17
    at IncomingMessage.<anonymous> (/usr/local/lib/node_modules/node.io/lib/node.io/request.js:384:17)
    at IncomingMessage.emit (events.js:81:20)
    at HTTPParser.onMessageComplete (http.js:133:23)
    at Socket.ondata (http.js:1227:22)
    at Socket._onReadable (net.js:683:27)
    at IOWatcher.onReadable [as callback] (net.js:177:10)
```

I have run npm install jquery; node.io is installed globally while all the other modules are installed in my home directory.

Installed modules:

``` bash
$ npm list
/Users/peter
├── coffee-script@1.1.1 
├── htmlparser@1.7.3 
├── jquery@1.6.2 
├─┬ jsdom@0.2.0 
│ ├── htmlparser@1.7.3 
│ └── request@1.9.8 
├─┬ node.io@0.3.3 -> /usr/local/lib/node_modules/node.io
│ ├── coffee-script@1.1.1 
│ ├── htmlparser@1.7.3 
│ └─┬ jquery@1.6.2 
│   └─┬ jsdom@0.2.0 
│     └── request@1.9.8 
└── request@1.9.8 
```

and

``` bash
$ npm list -g
/usr/local/lib
├─┬ node.io@0.3.3 
│ ├── coffee-script@1.1.1 
│ ├── htmlparser@1.7.3 
│ └─┬ jquery@1.6.2 
│   └─┬ jsdom@0.2.0 
│     └── request@1.9.8 
└─┬ npm@1.0.15 
  ├── abbrev@1.0.3 
  ├── node-uuid@1.2.0 
  ├── nopt@1.0.4 
  └── semver@1.0.7 
```

Please, do you have any idea what is wrong?

**Comment by pbowyer** [comment]
When I console.log e, the exception message is "Cannot find module 'jsdom'" - yet it appears to be installed?

**Comment by pbowyer** [comment]
A follow-up: I got past this error by playing with which packages were installed locally and remotely, to end up with the following:

``` bash
$ npm list
/Users/peter
├── coffee-script@1.1.1 
├── htmlparser@1.7.3 
├── jquery@1.6.2 
├─┬ jsdom@0.2.0 
│ ├── htmlparser@1.7.3 
│ └── request@1.9.8 
├─┬ node.io@0.3.3 -> /usr/local/lib/node_modules/node.io
│ ├── coffee-script@1.1.1 
│ ├── htmlparser@1.7.3 
│ └─┬ jquery@1.6.2 
│   └─┬ jsdom@0.2.0 
│     └── request@1.9.8 
└── request@1.9.8
```

And

``` bash
$ npm list -g
/usr/local/lib
├─┬ jsdom@0.2.0 
│ ├── htmlparser@1.7.3 
│ └── request@1.9.8 
├─┬ node.io@0.3.3 
│ ├── coffee-script@1.1.1 
│ ├── htmlparser@1.7.3 
│ └─┬ jquery@1.6.2 
│   └─┬ jsdom@0.2.0 
│     └── request@1.9.8 
└─┬ npm@1.0.15 
  ├── abbrev@1.0.3 
  ├── node-uuid@1.2.0 
  ├── nopt@1.0.4 
  └── semver@1.0.7
```

Once that was fixed however I encountered a further issue - valid jQuery that would run in my browser wouldn't execute correctly and I had to rewrite my code. Specifically:

``` javascript
            $('.listing').each(function (i, listing) {
                var project = {};
                project.date = $('.date', listing);
                projects.push(project);
            });
```

The third line didn't work and had to be rewritten as `project.date = $(listing).find('.date').text().trim();`, in order to prevent the following error:

``` bash
$ node.io -g nuffield
DEBUG: Running 1 worker..
DEBUG: GET http://www.nuffieldtheatre.co.uk/events/category/C81/ (request 71623)
DEBUG:   | GET /events/category/C81/ HTTP/1.1
DEBUG:   | Accept: */*
DEBUG:   | Accept-charset: ISO-8859-1,utf-8;q=0.7,*;q=0.3
DEBUG:   | User-agent: node.io
DEBUG:   | Host: www.nuffieldtheatre.co.uk
DEBUG: 200 http://www.nuffieldtheatre.co.uk/events/category/C81/ (response 71623)
DEBUG:   | Date: Sun, 17 Jul 2011 19:50:23 GMT
DEBUG:   | Server: Apache/2.0.54
DEBUG:   | X-powered-by: PHP/5.2.14
DEBUG:   | Expires: Mon, 26 Jul 1997 05:00:00 GMT
DEBUG:   | Pragma: no-cache
DEBUG:   | Set-cookie: exp_last_visit=995597423; expires=Mon, 16-Jul-2012 19:50:23 GMT; path=/,exp_last_activity=1310957423; expires=Mon, 16-Jul-2012 19:50:23 GMT; path=/,exp_tracker=a%3A1%3A%7Bi%3A0%3Bs%3A21%3A%22%2Fevents%2Fcategory%2FC81%2F%22%3B%7D; path=/
DEBUG:   | Last-modified: Sun, 17 Jul 2011 19:50:23 GMT
DEBUG:   | Vary: Accept-Encoding
DEBUG:   | Connection: close
DEBUG:   | Transfer-encoding: chunked
DEBUG:   | Content-type: text/html
{ '0': 
   { _attributes: 
      { _nodes: {},
        _nsStore: {},
        length: 0,
        _ownerDocument: [Object],
        _readonly: false,
        _parentNode: [Circular] },
     _ownerDocument: 
      { _childNodes: [Object],
        _ownerDocument: [Circular],
        _attributes: [Object],
        style: [Object],
        _childrenList: [Object],
        _childNodesList: [Object],
        _tagName: '#document',
        _nodeName: '#document',
        _contentType: 'text/html',
        _doctype: [Object],
        _implementation: [Object],
        _documentElement: [Object],
        _ids: [Object],
        _readonly: false,
        _URL: '/usr/local/lib/node_modules/node.io/lib/node.io/dom.js',
        _documentRoot: '/usr/local/lib/node_modules/node.io/lib/node.io',
        _queue: [Object],
        readyState: 'complete',
        _version: 1337,
        location: [Object],
        _parentWindow: [Object] },
     _childNodes: [ [Object] ],
     style: { position: 'st
/usr/local/lib/node_modules/node.io/node_modules/jquery/dist/node-jquery.js:92
    userAgent = navigator.userAgent,
                      ^
TypeError: Cannot read property 'userAgent' of undefined
    at /usr/local/lib/node_modules/node.io/node_modules/jquery/dist/node-jquery.js:92:23
    at /usr/local/lib/node_modules/node.io/node_modules/jquery/dist/node-jquery.js:964:2
    at Function.create (/usr/local/lib/node_modules/node.io/node_modules/jquery/dist/node-jquery.js:8996:2)
    at /usr/local/lib/node_modules/node.io/lib/node.io/dom.js:61:37
    at Object.<anonymous> (/Users/peter/Sites/nodeio/nuffield.js:27:32)
    at Function.each (/usr/local/lib/node_modules/node.io/node_modules/jquery/dist/node-jquery.js:670:20)
    at [object Object].each (/usr/local/lib/node_modules/node.io/node_modules/jquery/dist/node-jquery.js:288:17)
    at [object Object].<anonymous> (/Users/peter/Sites/nodeio/nuffield.js:25:27)
    at [object Object].parseHtml (/usr/local/lib/node_modules/node.io/lib/node.io/dom.js:66:18)
    at [object Object].<anonymous> (/usr/local/lib/node_modules/node.io/lib/node.io/request.js:108:18)
```

It's a very strange error message to have, and I don't understand why this jQuery should cause it to happen.

**Comment by pbowyer** [comment]
Source code for the scraper @ https://gist.github.com/1088002

**Comment by chriso** [comment]
The original error was due to [this line](https://github.com/chriso/node.io/commit/60147773e8f48465f8f0d77df2150f206bf3f6c3#L0L54). I didn't pick it up because I had jsdom installed globally! I've added jsdom as a required package in the latest release so this should be fixed.

The second bug should be fixed in [this commit](https://github.com/chriso/node.io/commit/d7b912e28d614dae89e453312de209c32139a66c#L1R57).

Let me know if you still have any issues.
